packages/metro-file-map/src/__tests__/HasteFS-test.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ * @format
+ */
+
+import HasteFS from '../HasteFS';
+
+describe('matchFilesWithContext', () => {
+  it(`matches files against context`, () => {
+    const hfs = new HasteFS({
+      rootDir: '/',
+      files: new Map([]),
+    });
+
+    // $FlowFixMe: mocking files
```

Comment:
Because I want the test to be focused on `matchFilesWithContext` and this is the path of least resistance. Going through the constructor means the mock paths will be resolved against the file system.

I'd need to mock out path to skip this resolution, which imo creates a more confusing test.